frontend/src/features/missions/MissionForm/ActionForm/ControlForm/index.tsx

```diff
@@ -199,18 +199,67 @@ export function ControlForm({
       return
     }
     setFieldValue(`envActions[${envActionIndex}].reportingIds`, [reportingId])
-    const reportingToAttachIndex = attachedReportings?.findIndex(reporting => reporting.id === reportingId)
-
-    if (reportingToAttachIndex !== -1) {
-      setFieldValue(`attachedReportings[${reportingToAttachIndex}].attachedEnvActionId`, currentAction?.id)
-    }
 
     const reportingToDetachIndex = attachedReportings?.findIndex(
       reporting => reporting.attachedEnvActionId === currentAction?.id && reporting.id !== reportingId
     )
     if (reportingToDetachIndex !== -1) {
       setFieldValue(`attachedReportings[${reportingToDetachIndex}].attachedEnvActionId`, undefined)
     }
+
+    const reportingToAttachIndex = attachedReportings?.findIndex(reporting => reporting.id === reportingId)
+    if (reportingToAttachIndex !== -1) {
+      setFieldValue(`attachedReportings[${reportingToAttachIndex}].attachedEnvActionId`, currentAction?.id)
+      // prefill infractions with the reporting details
+      const reporting = attachedReportings[reportingToAttachIndex]
+      setFieldValue(`envActions[${envActionIndex}].controlPlans`, [
+        { subThemeIds: reporting?.subThemeIds, tagIds: [], themeId: reporting?.themeId }
+      ])
+
+      if (
+        reporting &&
+        reporting.targetType !== ReportingTargetTypeEnum.OTHER &&
+        reporting.targetDetails &&
+        reporting.targetDetails.length > 0
+      ) {
+        const updatedInfractions = reporting.targetDetails.map(target => {
+          switch (reporting.targetType) {
+            case ReportingTargetTypeEnum.VEHICLE:
+              return {
+                controlledPersonIdentity: target?.vesselName ?? target?.operatorName,
+                registrationNumber: target?.externalReferenceNumber,
+                ...(reporting.vehicleType === VehicleTypeEnum.VESSEL && {
+                  vesselSize: target?.size
+                })
+              }
+
+            case ReportingTargetTypeEnum.COMPANY:
+              return {
+                companyName: target?.operatorName,
+                controlledPersonIdentity: target?.vesselName
+              }
+
```

Comment:
J'aurais rajouté 
```suggestion
ReportingTargetTypeEnum.INDIVIDUAL:
```
pour rendre explicite le fait que le default case correspond à ce cas.

Reply:
ok, je le fais dans mon autre PR en cours